### PR TITLE
feat(auth-fow-wallet): CHK-2872 add common pagopa IO auth

### PR DESCRIPTION
**Pull request by infantesimone**
### Motivation and context

The goal of this PR is to update authentication process for `eCommerce` and` payment wallet pagoPA `according new pagoPA platform auth session, in order to standardize the authentication process between all pagoPA services and IO App, for more [details](https://pagopa.atlassian.net/wiki/spaces/PPR/pages/1008336988/HowTo+call+pagoPA+platform+service+via+AppIO).
 Pay attention for new openapi:
- https://editor.swagger.io/?url=https://raw.githubusercontent.com/pagopa/pagopa-infra/ac626f6736c9bc9a78771d458df9e105fdf25d58/src/domains/ecommerce-app/api/ecommerce-for-io/v1/_openapi.json.tpl
- https://editor.swagger.io/?url=https://raw.githubusercontent.com/pagopa/pagopa-infra/ac626f6736c9bc9a78771d458df9e105fdf25d58/src/domains/pay-wallet-app/api/io-payment-wallet/v1/_openapi.json.tpl

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/pay-wallet-app/api/io-payment-wallet/v1/_openapi.json.tpl [hunk 1]
```diff
@@ -0,0 +1,944 @@
...
+          "max": {
+            "type": "integer",
+            "format": "int64",
+            "minimum": 0
+          }
+        }
+      }
+    },
+    "securitySchemes": {
+      "bearerAuth": {
```
Comment (pietro-tota) [inline src/domains/pay-wallet-app/api/io-payment-wallet/v1/_openapi.json.tpl:937]:
```suggestion
      "pagoPAPlatformSessionToken": {
```
I liked the `pagoPAPlatformSessionToken` auth token name being explicitly set as bearer token name in openapi as done for other openapi :D

File: src/domains/ecommerce-app/api/ecommerce-for-io/v1/_openapi.json.tpl [hunk 2]
```diff
@@ -0,0 +1,2250 @@
...
+            "description": "Invalid input id",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ProblemJson"
+                }
+              }
+            }
+          },
+          "404": {
```
Comment (ciuffagianluca) [inline src/domains/ecommerce-app/api/ecommerce-for-io/v1/_openapi.json.tpl:724]: Maybe in this api a `401` error code is missing.

File: src/domains/pay-wallet-app/api/io-payment-wallet/v1/_openapi.json.tpl [hunk 3]
```diff
@@ -0,0 +1,944 @@
...
+            "description": "Invalid input id",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ProblemJson"
+                }
+              }
+            }
+          },
+          "403": {
```
Comment (ciuffagianluca) [inline src/domains/pay-wallet-app/api/io-payment-wallet/v1/_openapi.json.tpl:425]: `401` error code is missing in this api

**Comment by ciuffagianluca** [comment]
[Non blocking comment] I saw that that the `start-session` in ecommerce-for-io is not present in the new openapi. This means that it will be managed out of the user payment flow?

**Review by ciuffagianluca — DISMISSED** [review]

File: src/domains/pay-wallet-app/api/io-payment-wallet/v1/_openapi.json.tpl (hunk 1 above)
Comment (infantesimone) [reply src/domains/pay-wallet-app/api/io-payment-wallet/v1/_openapi.json.tpl:966]: great!

**Pull request merged**